### ISSUE TITLE
Support for schedulers with extra configurations, e.g. `ReduceLROnPlateau`

### ISSUE THREAD

**Issue by surajpaib** [ibro45/lightningbringer#4]
**Problem faced**
Some schedulers require extra configuration in pytorch lightning, for example, `ReduceLROnPlateau` where an additional `monitor` argument needs to be passed to the `schedulers` config

**Example**
Sample config for such scheduler
```
    schedulers:
      scheduler:
        _target_: torch.optim.lr_scheduler.ReduceLROnPlateau
      interval: epoch
      monitor: val/loss
      strict: True
```

**Why does it not work**
While instantiating the schedulers, we pass optimizer object to the first-level of the config and it does not pass on to `schedulers.scheduler` which throws an error. 

**Solution**
Instantiating schedulers like below,

```
   system.schedulers = instantiate(conf.system.schedulers,
                                    _convert_="all",
                                    scheduler={"optimizer": system.optimizers})
```

However, this needs to work for both the general case and this specific case

**Comment by ibro45** [comment]
f60bed41c679430e3ec1ba2ea528360d2ea56047

Thanks, I just added the piece of code you wrote. Once https://github.com/facebookresearch/hydra/issues/1758 is resolved, we won't need any of these workarounds.

**Comment by ibro45** [comment]
remember this shit? @surajpaib 🤣
